service/src/test/java/bio/terra/workspace/serdes/ControlledGcsBucketResourceTest.java

```diff
@@ -21,15 +21,19 @@ public void testValidateThrows() {
     assertThrows(
         MissingRequiredFieldException.class,
         () ->
-            ControlledResourceFixtures.makeDefaultControlledGcsBucketResource()
-                .workspaceId(null)
+            ControlledGcsBucketResource.builder()
```

Comment:
So in this class we validate required fields in the builder, whereas in some others this happens in the constructor. I'd do it in the builder everywhere.

Reply:
I don't think so. The build is calling the constructor and the constructor is doing the validation. We do validation in all of the resource constructors.